src/main/java/com/dezorganizacja/bridge/view/text/TerminalApplication.java

```diff
@@ -0,0 +1,64 @@
+package com.dezorganizacja.bridge.view.text;
+
+import com.dezorganizacja.bridge.presenter.MainPresenter;
+
+import java.util.ArrayList;
+
+import java.util.Observable;
+import java.util.Observer;
+import java.util.Scanner;
+
+/**
+ * Created by maciek on 13.04.15.
+ */
+public class TerminalApplication implements Observer {
+    private static MainPresenter mainPresenter;
+    private OrdersHandler handler = new OrdersHandler ();
+    private String state;
+
+    public static void setMainPresenter(MainPresenter mainPresenter) {
+        TerminalApplication.mainPresenter = mainPresenter;
+    }
+
+    public void setState(String state) {
+        this.state = state;
+    }
+
+    public void launch () {
+        TerminalApplication.mainPresenter.addObserver(this);
+        this.showWelcome();
+        this.setState("loading");
+        this.startHandling ();
+    }
+
+    public void update(Observable observable, Object o) {
+        this.changeState();
+    }
+
+    public void startHandling () {
+        Scanner input = new Scanner(System.in);
+        handler.dispatchOrder("menu", new ArrayList<>());
```

Comment:
Dobrze, ale masz metodę showWelcome. dlaczego np tam albo bezpośrednio w launch tego nie ma a dopiero w startHandling? Jeśli mam metodę showWelcome to spodziewam się żę to wypisuje całość powitania a nie tylko jakiś fragment a potem się wypisuje reszta w jakiejś innej metodzie.